File: Data Structure/Hash/Easy/LET_no170_Two Sum III - Data structure design.cpp
class TwoSum {
private:
    unordered_map<long, int> memo;
public:
    TwoSum() { }
    
    void add(int number) {
        memo[long(number)]++;
    }
    
    bool find(int value) {
        for (auto it : memo) {
            long cur = it.first;
            int count = it.second;
            long target = long(value) - cur;
            if (memo.count(target) && (target != cur || count > 1))
                return true;
        }

        return false;
    }
};

/**
 * Your TwoSum object will be instantiated and called as such:
 * TwoSum* obj = new TwoSum();
 * obj->add(number);
 * bool param_2 = obj->find(value);
 */

File: Data Structure/Hash/Easy/LET_no170_Two Sum III - Data structure design.py
class TwoSum:

    def __init__(self):
        self.memo = collections.defaultdict(int)

    def add(self, number: int) -> None:
        self.memo[number] += 1

    def find(self, value: int) -> bool:
        for key in self.memo:
            if value - key in self.memo and (value - key != key or self.memo[value - key] > 1):
                return True
        
        return False



# Your TwoSum object will be instantiated and called as such:
# obj = TwoSum()
# obj.add(number)
# param_2 = obj.find(value)